src/python/gem5/prebuilt/riscvunleashed/riscvunleashed_board.py

```diff
@@ -0,0 +1,573 @@
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import os
+from typing import (
+    List,
+    Optional,
+)
+
+import m5
+from m5.objects import (
+    AddrRange,
+    BadAddr,
+    Bridge,
+    CowDiskImage,
+    Frequency,
+    HiFive,
+    IGbE_e1000,
+    IOXBar,
+    PMAChecker,
+    Port,
+    RawDiskImage,
+    RiscvBootloaderKernelWorkload,
+    RiscvMmioVirtIO,
+    RiscvRTC,
+    VirtIOBlock,
+    VirtIORng,
+)
+from m5.util.fdthelper import (
+    Fdt,
+    FdtNode,
+    FdtProperty,
+    FdtPropertyStrings,
+    FdtPropertyWords,
+    FdtState,
+)
+
+from gem5.components.boards.abstract_system_board import AbstractSystemBoard
+from gem5.components.boards.kernel_disk_workload import KernelDiskWorkload
+from gem5.components.boards.se_binary_workload import SEBinaryWorkload
+from gem5.components.cachehierarchies.classic.private_l1_private_l2_walk_cache_hierarchy import (
+    PrivateL1PrivateL2WalkCacheHierarchy,
+)
+from gem5.components.memory import SingleChannelDDR4_2400
+from gem5.components.processors.abstract_processor import AbstractProcessor
+from gem5.components.processors.cpu_types import CPUTypes
+from gem5.isas import ISA
+from gem5.resources.resource import AbstractResource
+from gem5.utils.override import overrides
+from gem5.utils.requires import requires
+
+from .riscvunleashed_cache import RISCVUnleashedCacheHierarchy
+from .riscvunleashed_processor import U54Processor
+
+
+class RiscvUnleashedBoard(AbstractSystemBoard, KernelDiskWorkload):
+    """
+    A board capable of full system simulation for RISC-V.
+
+    At a high-level, this is based on the HiFive Unmatched board from SiFive.
+
+    This board assumes that you will be booting Linux. The boards caches,
+
+    memory hierarchy has been validated against the Sifive HiFive Unleashed
+
+    Board for SPEC 2017 benchmark suite
+
+    **Limitations**
+    * Only works with classic caches
+    """
+
+    def __init__(
+        self,
+    ) -> None:
+        clk_freq = "1GHz"
+        cache_hierarchy = RISCVUnleashedCacheHierarchy(l2_size="2MiB")
+        processor = U54Processor()
+        memory = SingleChannelDDR4_2400("8GB")
```

Comment:
Specify that you want a power of 2, not a power of 10.
```suggestion
        memory = SingleChannelDDR4_2400("8GiB")
```